contracts/NonfungiblePositionManager.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {ERC721} from "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
```

Comment:
Copied ERC721Permit from v3. Stuck with OZ for now since they have helpful modifiers and their _beforeTokenTransfer is a bit easier to use than overriding transfer